packages/lisk-transactions/src/utils/sign_and_verify.ts

```diff
@@ -84,51 +85,74 @@ export const verifyMultisignatures = (
 			errors: [new TransactionError('Sender does not have valid multimin', id)],
 		};
 	}
-	if (!signatures || (signatures && signatures.length < minimumValidations)) {
+	if (!Array.isArray(signatures)) {
 		return {
 			verified: false,
 			errors: [new TransactionError('Missing signatures', id, '.signatures')],
 		};
 	}
+	if (signatures.length > MAX_MULTISIG_SIGNATURES) {
+		return {
+			verified: false,
+			errors: [
+				new TransactionError(
+					'Exceeds maximum of 15 signatures.',
+					id,
+					'.signatures',
+				),
+			],
+		};
+	}
 
-	const verifiedSignatures: string[] = [];
-	const checkedPublicKeys: string[] = [];
+	const checkedPublicKeys = new Map();
```

Comment:
why do we need to use `Map` here? also, it's looping thorough the memerPublicKeys once, so im not sure if it's needed

Reply:
if it's needed, then `Set` would be better

Reply:
We need `checkedPublicKeys` due to [cryptographic issue](https://github.com/LiskHQ/lisk/issues/2540) where multiple signatures can be created from the same key. For example, if one key verifies a signature, then it is possible for it to verify another signature as well. Having `checkedKeys` will avoid single key from verifying multiple signature. I will add a comment indicating the reason for clarity.

Reply:
Instead of using `Map` I think `Set` should be fine.
because we are looping through public keys once, 
`memberPublicKeys.forEach(publicKey => {`
there is no possibility of checking the same public key twice with 2 different public key?